packages/app/src/systems/Chains/eth/hooks/useManageEthAssets.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+import { isAddress } from 'viem';
+import { useToken } from 'wagmi';
+
+import { useAssets } from './useAssets';
+
+export const useManageEthAssets = (newAssetAddress: string) => {
+  const { assets, handlers } = useAssets();
+
+  const { data, isError, isLoading } = useToken({
+    address: newAssetAddress as `0x${string}`,
+  });
+
+  const { filteredAssets, doesAssetExist } = useMemo(() => {
+    const isValidAddress = isAddress(newAssetAddress);
+    if (isValidAddress) {
+      const filteredAssets = assets.filter(
+        (asset) => asset.address === newAssetAddress
+      );
+      return { filteredAssets, doesAssetExist: filteredAssets.length };
+    }
+    const newAssetsArray = assets.filter((asset) =>
+      asset.symbol?.toLowerCase().startsWith(newAssetAddress.toLowerCase())
+    );
+    if (!newAssetsArray.length) {
+      return { filteredAssets: assets, doesAssetExist: false };
+    }
+    return { filteredAssets: newAssetsArray, doesAssetExist: true };
+  }, [assets, newAssetAddress]);
+
+  return {
+    assets: filteredAssets,
+    handlers,
+    showUseTokenButton: !isError && !!data && !doesAssetExist,
+    showCustomTokenButton:
+      isError &&
+      !!newAssetAddress.length &&
+      isAddress(newAssetAddress) &&
+      !doesAssetExist,
+    assetInfo: data,
+    isLoading: isLoading && isAddress(newAssetAddress) && !doesAssetExist,
```

Comment:
should be only `isLoading`

the reason is that we wanna see it wrong in the screen if `useToken` is loading all the time / unnecessarily